src/extension.ts

```diff
@@ -255,6 +255,28 @@ export class Extension implements vscode.Disposable {
     return this.enabled = enabled
   }
 
+  /**
+   * Make all selections in the editor non-empty by selecting at least one character.
+   */
+  normalizeSelections(editor: vscode.TextEditor) {
+    if (this.modeMap.get(editor.document) !== Mode.Normal) { return; }
+    if (this.configuration.get('selections.allowEmpty')) { return; }
+    if (editor.selections.some(sel => sel.isEmpty)) {
```

Comment:
Since this is called every time the selection changes, I'd like to avoid allocations as much as possible. Could you rewrite this so it only uses one loop and no `map` / `some`? For instance, create an array and populate it manually with a `for` loop, and use a boolean to know whether some changes were applied (i.e `selection.isEmpty` was true at some point). If that boolean is true at the end of the loop, set `editor.selections` to the array you built.

Reply:
Done.